packages/terra-date-input/src/DateInput.jsx

```diff
@@ -264,6 +266,17 @@ class DateInput extends React.Component {
     }
   }
 
+  /**
+   * Takes a key press from the day and year input, and processes it based on the value of the keycode
+   * Prevents non-numeric characters from being entered in Safari browser.
+   * @param {Object} event Event object generated from the event delegation.
+   */
+  handleKeyPress = event => {
+    const keyCode = (typeof event.which === 'undefined') ? event.keyCode : event.which;
+    const input = String.fromCharCode(keyCode);
+    if (!input.match(/^[0-9]+$/) && !event.metaKey) event.preventDefault();
+  }
```

Comment:
both event.keyCode and event.which are deprecated : 
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode and https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/which 

So instead of `keyCode` and `which` can we use [key](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key) property of keyboard event object. which returns the actual value of the key pressed instead of unicode value. This eliminates need of converting unicode value to string again for comparsion with regex.


```suggestion
  handleKeyPress = event => {
    const input = event.key;
    if (!input.match(/^[0-9]+$/) && !event.metaKey) event.preventDefault();
  }
```